### ISSUE TITLE
Changing learning rate while training with nnx interface

### ISSUE THREAD

**Issue by segasai** [google/flax#4531]
Hi,

I'm struggling to figure out how to implement  the adjustment of the learning rate while training (i.e. for   LRReduceOnPlateau-like scheduling)  for the new nnx interface.
Previously when using the linen interface I used the transform_state that passed to apply_gradients() call of the train_state. But I am a bit lost how do it with nnx and if there is an easier way to do it than declaring your own TrainState class. 

I.e. for this example from the documentation: 

```python
from flax import nnx
import optax


class Model(nnx.Module):
  def __init__(self, din, dmid, dout, rngs: nnx.Rngs):
    self.linear = nnx.Linear(din, dmid, rngs=rngs)
    self.linear_out = nnx.Linear(dmid, dout, rngs=rngs)

  def __call__(self, x):
    x = nnx.relu(self.linear(x))
    return self.linear_out(x)

model = Model(2, 64, 3, rngs=nnx.Rngs(0))  # eager initialization
optimizer = nnx.Optimizer(model, optax.adam(1e-3))  # reference sharing

@nnx.jit  # automatic state management for JAX transforms
def train_step(model, optimizer, x, y):
  def loss_fn(model):
    y_pred = model(x)  # call methods directly
    return ((y_pred - y) ** 2).mean()

  loss, grads = nnx.value_and_grad(loss_fn)(model)
  optimizer.update(grads)  # in-place updates

  return loss

loss1  = train_model(model, optimizer, x,y)
# how do I reduce the learning rate here ? 
loss2 = train_model(model, optimizer, x,y)

```
How do I change the learning rate between train_model calls ? 

Thanks! 

EDIT: I've tried to multiply the grads by the scaling factor in optimizer.update() but that does not work -- it complains about the 'unsupported operand types State and DynamicJaxprTracer.

**Comment by segasai** [comment]
Closing the issue as replacing 

```
optimizer = nnx.Optimizer(model, optax.adam(1e-3))  
```
by 
```
optimizer = nnx.Optimizer(model,  optax.chain(
        optax.adam(1e-3),
        optax.contrib.reduce_on_plateau(
            patience=10,
            cooldown=1,
            factor=0.5,
            rtol=1e-5,
        ))
```

as well as replacing optimizer.update(grads) by optimizer.update(grads, value=loss) did the trick
